pingcap/tidb
# session: remove distributed tasks limitation from upgrade

### What problem does this PR solve?


Issue Number: close #54061

Problem Summary:

> During the rolling upgrade, when the distributed execution framework is enabled, the add index task is processed, and data indexes may be inconsistent.

The limitation is only a defensive measure based on previous issues, we don't have unsolved data inconsistent issues.

Since the issue is fixed(https://github.com/pingcap/tidb/issues/50895), and the limitation introduces inconvenience to many users, we should remove the check in the following versions.

### What changed and how does it work?

Remove distributed task configuration check before upgrade.

### Check List

Tests 

  Before this PR:
  1. Deploy a nightly cluster without TiDB instance.
  2. Start TiDB v7.5.
  3. Set `@@global.tidb_enable_dist_task` to `ON`.
  4. Kill TiDB v7.5.
  5. Start TiDB master.
  TiDB cannot startup and report error:
  ```
  [FATAL] [bootstrap.go:1495] ["cannot upgrade when tidb_enable_dist_task is enabled, please set tidb_enable_dist_task to off before upgrade"] [] [stack="github.com/pingcap/tidb/pkg/session.checkDistTask\n\t/home/tangenta/tidb/pkgsession/bootstrap.go:1495\ngithub.com/pingcap/tidb/pkg/session.upgrade\n\t/home/tangenta/tidb/pkg/session/bootstrap.go:1538\ngithub.com/pingcap/tidb/pkg/session.runInBootstrapSession\n\t/home/tangenta/tidb/pkg/session/sessiongo:3738\ngithub.com/pingcap/tidb/pkg/session.bootstrapSessionImpl\n\t/home/tangenta/tidb/pkg/session/session.go:3460\ngithub.com/pingcap/tidb/pkg/session.BootstrapSession\n\t/home/tangenta/tidb/pkg/session/session.go:3406\nmaincreateStoreDDLOwnerMgrAndDomain\n\t/home/tangenta/tidb/cmd/tidb-server/main.go:418\nmain.main\n\t/home/tangenta/tidb/cmd/tidb-server/main.go:321\nruntime.main\n\t/home/tangenta/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4linux-amd64/src/runtime/proc.go:272"]
  ```
  After this PR, we removed the check `checkDistTask`, TiDB can startup and upgrade normally:
  ```
  mysql> select tidb_version();
  +-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
  | tidb_version()                                                                                                                                                                                                  |
  +-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
  | Release Version: v8.4.0-this-is-a-placeholder
  Edition: Community
  Git Commit Hash: None
  Git Branch: None
  UTC Build Time: None
  GoVersion: go1.23.2
  Race Enabled: false
  Check Table Before Drop: false
  Store: tikv |
  +-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
  1 row in set (0.00 sec)
  
  mysql> select @@global.tidb_enable_dist_task;
  +--------------------------------+
  | @@global.tidb_enable_dist_task |
  +--------------------------------+
  |                              1 |
  +--------------------------------+
  1 row in set (0.00 sec)
  ```
  > - [ ] I checked and no code files have been changed.
  > 

Side effects


Documentation


### Release note



Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```
